### PR TITLE
feat: type improvements

### PR DESCRIPTION
**Pull request by danielroe**
Briefly:
* move whole project to strict typing
* fully type mixins/components
* address any issues flagged in the process

### CODE REVIEW

File: src/runtime/components/nuxt-img.vue [hunk 1]
```diff
@@ -34,7 +42,8 @@ export default {
       }
       return this.$img(this.src, this.nModifiers, this.nOptions)
     },
-    nSizes () {
+    /* eslint-disable no-undef */
+    nSizes (): ReturnType<typeof $Img['getSizes']> {
```
Comment (danielroe) [inline src/runtime/components/nuxt-img.vue:46]: We could also import the type directly. It will be excluded either way. Thoughts?

Comment (pi0) [reply src/runtime/components/nuxt-img.vue:46]: I think while we are moving towards strict typing, makes sense to also care about naming internal interfaces

File: src/runtime/components/image.mixin.ts [hunk 2]
```diff
@@ -25,41 +30,52 @@ export const imageMixin = {
...
         longdesc: this.longdesc,
         ismap: this.ismap,
-        crossorigin: this.crossorigin,
+        crossorigin: this.crossorigin === true ? 'anonymous' : this.crossorigin || undefined,
         loading: this.loading,
         decoding: this.decoding
       }
     },
-    nModifiers () {
+    nModifiers (): { width?: number, height?: number, format?: string, quality?: string | number, background?: string, fit?: string } & Record<string, any> {
```
Comment (pi0) [inline src/runtime/components/image.mixin.ts:63]: This return type (and rest of the functions) are already inferred by typescript. (also mkdist at least currently for some reason is not preserving types at all)

File: src/runtime/components/nuxt-img.vue (hunk 1 above)
Comment (pi0) [reply src/runtime/components/nuxt-img.vue]: 892b3f9509b80727803a995dec282c62a6d2f511

**Review by pi0 — APPROVED** [review]

**Pull request merged**